### ISSUE TITLE
How to consider the mineral particle composition using HYD.ExtData in OSOAA

### ISSUE THREAD

**Issue by yichen-li2016** [CNES/RadiativeTransferCode-OSOAA#21]
Hi,

I have recently used OSOAA to simulate the effect of the content of different constituents in the sea on the polarization signal. I divided the water column into two components: phytoplankton and mineral particles, calculated the number of particles of different components based on a power-law model, and obtained the phase matrix by Mie theory calculations. The phase matrix and cross-sectional area of these particles are mixed using the equations on pages 81-82 of the user manual. I encountered the following problems:

When I set the HYD mode to 2, the program will set the Csed to 0 by default, and when the Csed is not 0, the result I get is Nan. In fact, the HYD.ExtData has included the scattering phase function of inorganic particles. However, setting the CHL concentration alone cannot accurately express inorganic particles.  What should I do?

Best regards,

Qiang

**Comment by yichen-li2016** [comment]
What are the functions of EXTINCTION_COEF and SCATTERING_COEF in ExtData? I tried different EXTINCTION_COEF and SCATTERING_COEF with the same scattering phase function and got the same result.

**Comment by blafranc** [comment]
Dear Qiang,

Indeed, the case HYD.Mod 2 allows using a user-defined phase matrix, but this one is not ascribed to “mineral-like” particle component of the OSOAA model; it is ascribed to the phytoplankton component only. Consequently, for this use case, the marine profile is only defined for the phytoplankton.

Actually, your case study does not require the use of the option "HYD.ExtData" of OSOAA since you use Mie theory to calculate the Mueller Matrix. Therefore, you need to select the option HYD.Mod 1 for which the size distribution of both phytoplankton and mineral particles and their concentration is a Junge power law model

Please note the version 2.0 of the OSOAA model will be available in few weeks. It will make possible to introduce a user-defined external Mueller matrix of hydrosols, coupled with a user’s marine profile of absorption and scattering coefficients. This use case will be activated by the new option HYD.Model 3. Therefore, it will be possible to consider the global effect of both phytoplankton and “mineral-like” particles in the marine profile and in the Mueller matrix (based on user’s measurements or based on simulations).
This means that for your case study, you will only need to complete your phase matrix with a profile of absorption and scattering coefficients.

Best regards

**Comment by blafranc** [comment]
The  EXTINCTION_COEF and SCATTERING_COEF values in the user-defined file are used to calculate the single scattering albedo, defined as the ratio SCATTERING_COEF / EXTINCTION_COEF.
That is why the unit you use for these coefficients is not important. 
Best regards

**Comment by yichen-li2016** [comment]
Dear Lafrance,

Thank you so much for your reply, your reply is crucial to me! I've struggled with this for weeks!
The good news for OSOAA v2.0 is exciting!

I obtained identical results by adjusting different single scattering albedos by modifying various EXTINCTION_COEF and SCATTERING_COEF, given the same scattering phase function. In the two simulations mentioned above, different filenames for ExtData were used to ensure that OSOAA reruns.

Best regards,
Qiang
